### PR TITLE
doc: Rework the tutorial

### PR DESCRIPTION
**Pull request by kgriffs**
Rework the tutorial to be more consistent, and to better explain certain principles that drive Falcon's design. Also optimize the logo image, and tweak a few other parts of the docs and related to the tutorial.

### CODE REVIEW

File: docs/user/tutorial.rst [hunk 1]
```diff
@@ -85,98 +93,164 @@ Note the method signature. ``env`` and ``start_response`` are standard
...
-In Falcon, you map incoming requests to things called "Resources". A
-Resource is just a regular Python class that includes some methods that
-follow a certain naming convention. Each of these methods corresponds to
-an action that the API client can request be performed in order to fetch
-or transform the resource in question.
+Falcon uses Python classes to represent resources. These classes
+act as controllers in your application, converting incoming requests
+into a series of internal actions to be passed off to other components
+or backend services, collecting the results, and then finally using
+those results to compose a response back to the client.
```
Comment (jmvrbanac) [inline docs/user/tutorial.rst]: That's a long sentence ;)

File: docs/user/tutorial.rst [hunk 2]
```diff
@@ -186,99 +260,170 @@ params, as we shall see later on.
...
+    api.add_route('/images', images)
 
-Note the use of ``resp.data`` in lieu of ``resp.body``. If you assign a
-bytestring to the latter, Falcon will figure it out, but you can
-get a little performance boost by assigning directly to ``resp.data``.
+Now, when a request comes in for ``/images``, Falcon will call the
+responder on the images resource that corresponds to the requested
+HTTP method.
+
+Let's try it. Restart Gunicorn (unless using ``--reload``), and
```
Comment (jmvrbanac) [inline docs/user/tutorial.rst]: Should it be "unless you're using ``--reload``"?

Comment (kgriffs) [inline docs/user/tutorial.rst]: 👍

File: docs/user/tutorial.rst (hunk 1 above)
Comment (kgriffs) [inline docs/user/tutorial.rst]: Good catch! I attempted to simplify this paragraph. Let me know how the new version reads.

**Comment by kgriffs** [comment]
Rebased

**Pull request merged**